### ISSUE TITLE
controller-gen v0.11.3 panic while building rook

### ISSUE THREAD

**Issue by chenyuanrun** [rook/rook#13814]
controller-gen v0.11.3 panic while building rook

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

`make build` panic:
```
Updating CRD manifests
Generating ceph crds
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa05a3e]

goroutine 158 [running]:
go/types.(*Checker).handleBailout(0xc001594400, 0xc0014c5d40)
        /usr/local/go/src/go/types/check.go:367 +0x88
panic({0xbbf300?, 0x12a28a0?})
        /usr/local/go/src/runtime/panic.go:770 +0x132
go/types.(*StdSizes).Sizeof(0x0, {0xdb6fb8, 0x12ab0c0})
        /usr/local/go/src/go/types/sizes.go:228 +0x31e
go/types.(*Config).sizeof(...)
        /usr/local/go/src/go/types/sizes.go:333
go/types.representableConst.func1({0xdb6fb8?, 0x12ab0c0?})
        /usr/local/go/src/go/types/const.go:76 +0x9e
go/types.representableConst({0xdbd290, 0x12773f8}, 0xc001594400, 0x12ab0c0, 0xc0014c1b40)
        /usr/local/go/src/go/types/const.go:106 +0x2c7
go/types.(*Checker).representation(0xc001594400, 0xc0011fff40, 0x12ab0c0)
        /usr/local/go/src/go/types/const.go:256 +0x65
go/types.(*Checker).representable(0xc001594400, 0xc0011fff40, 0x12ab0c0)
        /usr/local/go/src/go/types/const.go:239 +0x26
go/types.(*Checker).shift(0xc001594400, 0xc0011ffec0, 0xc0011fff40, {0xdbb0d0, 0xc000da7320}, 0x14)
        /usr/local/go/src/go/types/expr.go:650 +0x1eb
go/types.(*Checker).binary(0xc001594400, 0xc0011ffec0, {0xdbb0d0, 0xc000da7320}, {0xdbb5e0, 0xc00060c540}, {0xdbb5e0, 0xc00060c560}, 0x14, 0x70c71)
        /usr/local/go/src/go/types/expr.go:796 +0x150
go/types.(*Checker).exprInternal(0xc001594400, 0x0, 0xc0011ffec0, {0xdbb0d0, 0xc000da7320}, {0x0, 0x0})
        /usr/local/go/src/go/types/expr.go:1416 +0x206
go/types.(*Checker).rawExpr(0xc001594400, 0x0, 0xc0011ffec0, {0xdbb0d0?, 0xc000da7320?}, {0x0?, 0x0?}, 0x0)
        /usr/local/go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).exprInternal(0xc001594400, 0x0, 0xc0011ffec0, {0xdbaf80, 0xc00060c580}, {0x0, 0x0})
        /usr/local/go/src/go/types/expr.go:1320 +0x178
go/types.(*Checker).rawExpr(0xc001594400, 0x0, 0xc0011ffec0, {0xdbaf80?, 0xc00060c580?}, {0x0?, 0x0?}, 0x0)
        /usr/local/go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).expr(0xc001594400, 0x0?, 0xc0011ffec0, {0xdbaf80?, 0xc00060c580?})
        /usr/local/go/src/go/types/expr.go:1513 +0x30
go/types.(*Checker).binary(0xc001594400, 0xc0011ffec0, {0xdbb0d0, 0xc000da7350}, {0xdbaf80, 0xc00060c580}, {0xdbb5e0, 0xc00060c5a0}, 0xd, 0x70c76)
        /usr/local/go/src/go/types/expr.go:783 +0xa5
go/types.(*Checker).exprInternal(0xc001594400, 0x0, 0xc0011ffec0, {0xdbb0d0, 0xc000da7350}, {0x0, 0x0})
        /usr/local/go/src/go/types/expr.go:1416 +0x206
go/types.(*Checker).rawExpr(0xc001594400, 0x0, 0xc0011ffec0, {0xdbb0d0?, 0xc000da7350?}, {0x0?, 0x0?}, 0x1)
        /usr/local/go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).use1(0xc001594400, {0xdbb0d0, 0xc000da7350}, 0x0)
        /usr/local/go/src/go/types/call.go:1043 +0x1a5
go/types.(*Checker).useN(...)
        /usr/local/go/src/go/types/call.go:1004
go/types.(*Checker).use(...)
        /usr/local/go/src/go/types/call.go:994
go/types.(*Checker).callExpr(0xc001594400, 0xc0011ffe00, 0xc000baef00)
        /usr/local/go/src/go/types/call.go:196 +0x1770
go/types.(*Checker).exprInternal(0xc001594400, 0x0, 0xc0011ffe00, {0xdbb6a0, 0xc000baef00}, {0x0, 0x0})
        /usr/local/go/src/go/types/expr.go:1374 +0xf8
go/types.(*Checker).rawExpr(0xc001594400, 0x0, 0xc0011ffe00, {0xdbb6a0?, 0xc000baef00?}, {0x0?, 0x0?}, 0x0)
        /usr/local/go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).expr(0xc001594400, 0xc0003a40e0?, 0xc0011ffe00, {0xdbb6a0?, 0xc000baef00?})
        /usr/local/go/src/go/types/expr.go:1513 +0x30
go/types.(*Checker).exprInternal(0xc001594400, 0x0, 0xc0011ffe00, {0xdbb040, 0xc000baef40}, {0x0, 0x0})
        /usr/local/go/src/go/types/expr.go:1190 +0x1f1f
go/types.(*Checker).rawExpr(0xc001594400, 0x0, 0xc0011ffe00, {0xdbb040?, 0xc000baef40?}, {0x0?, 0x0?}, 0x0)
        /usr/local/go/src/go/types/expr.go:979 +0x19e
go/types.(*Checker).expr(0xc001594400, 0x0?, 0xc0011ffe00, {0xdbb040?, 0xc000baef40?})
        /usr/local/go/src/go/types/expr.go:1513 +0x30
go/types.(*Checker).varDecl(0xc001594400, 0xc001194720, {0xc000b08878, 0x1, 0x1}, {0x0, 0x0}, {0xdbb040, 0xc000baef40})
        /usr/local/go/src/go/types/decl.go:521 +0x17b
go/types.(*Checker).objDecl(0xc001594400, {0xdc2a58, 0xc001194720}, 0x0)
        /usr/local/go/src/go/types/decl.go:194 +0x9e5
go/types.(*Checker).packageObjects(0xc001594400)
        /usr/local/go/src/go/types/resolver.go:693 +0x4dd
go/types.(*Checker).checkFiles(0xc001594400, {0xc000a8c000, 0x7, 0x7})
        /usr/local/go/src/go/types/check.go:408 +0x1a5
go/types.(*Checker).Files(...)
        /usr/local/go/src/go/types/check.go:372
sigs.k8s.io/controller-tools/pkg/loader.(*loader).typeCheck(0xc0003d90e0, 0xc000524880)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.3/pkg/loader/loader.go:286 +0x36a
sigs.k8s.io/controller-tools/pkg/loader.(*Package).NeedTypesInfo(0xc000524880)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.3/pkg/loader/loader.go:99 +0x39
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check(0xc000c11920, 0xc000524880)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.3/pkg/loader/refs.go:268 +0x2b7
sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check.func1(0x5a?)
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.3/pkg/loader/refs.go:262 +0x53
created by sigs.k8s.io/controller-tools/pkg/loader.(*TypeChecker).check in goroutine 1
        /root/go/pkg/mod/sigs.k8s.io/controller-tools@v0.11.3/pkg/loader/refs.go:260 +0x1c5
make[3]: *** [Makefile:203: crds] Error 2
```

**Expected behavior:**

`make build` do not panic

**How to reproduce it (minimal and precise):**

`make build` with helm v3.14.2

**File(s) to submit**:

None

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of kubectl commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook kubectl Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/kubectl-plugin)

**Environment**:
* OS (e.g. from /etc/os-release): almalinux 8.8 amd64
* Kernel (e.g. `uname -a`): 4.18.0-477.10.1.el8_8.x86_64
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): v1.13.5
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

**Comment by chenyuanrun** [comment]
I fix it with controller_gen v0.14.0

**Comment by subhamkrai** [comment]
we do have same v0.11.3 and it is working for us
https://github.com/rook/rook/blob/ca5972dece9243901dc9dd139c6536c1db518472/Makefile#L35
